scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,33 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+set +e
+dpkg-query -l opensearch 2>&1 | grep ^ii >/dev/null
+EX_CODE=$?
+if [ $EX_CODE -eq 0 ]; then
+  OPENSEARCH_ALREADY_INSTALLED=yes
+else
+  OPENSEARCH_ALREADY_INSTALLED=no
+fi
+set -e
+
```

Comment:
There is a better way to do a systemctl check instead of dpkg-query, as in the line 19 of the same file, that way is more accurate and less messy.

Reply:
done.